### PR TITLE
Unliftio migration and remove low-level streaming API

### PR DESCRIPTION
**Pull request by mtesseract**
Preperation for upcoming changes regarding concurrency.
This PR makes nakadi-client adopt the new unliftio foundation and removes the deprecated (by Nakadi) low-level streaming API. See https://zalando.github.io/nakadi/manual.html#/event-types/name/events_get.

### CODE REVIEW

**Comment by mtesseract** [comment]
@etorreborre, could you have a look when you find the time? :-)

File: tests/Network/Nakadi/EventTypes/Test.hs [hunk 1]
```diff
@@ -122,18 +142,30 @@ testEventTypePublishData conf = runApp . runNakadiT conf $ do
...
                               }
-  withAsync (delayedPublish Nothing [event]) $ \asyncHandle -> do
+  subscriptionId <- createMySubscription
+  batchTv <- newTVarIO Nothing
+  res <- try $ withAsync (delayedPublish Nothing [event]) $ \asyncHandle -> do
     liftIO $ link asyncHandle
-    (Just batchConsumed) :: Maybe (EventStreamBatch (DataChangeEvent Foo)) <-
-      eventsProcessConduit (Just consumeParametersSingle) myEventTypeName Nothing headC
-    liftIO $ isJust (batchConsumed^.L.events) @=? True
+    subscriptionProcess (Just consumeParametersSingle) subscriptionId (storeBatch batchTv)
```
Comment (etorreborre) [inline tests/Network/Nakadi/EventTypes/Test.hs]: It's not so trivial to test a subscription... But I totally understand what you are doing.

File: tests/Network/Nakadi/Examples/Subscription/Process.hs [hunk 2]
```diff
@@ -7,7 +7,7 @@ import           Data.Aeson
 import qualified Network.Nakadi as Nakadi
 import Network.Nakadi (MonadNakadi)
 import Control.Monad.Logger
-import Control.Monad.IO.Unlift
+import Control.Monad.Catch (MonadMask)
```
Comment (etorreborre) [inline tests/Network/Nakadi/Examples/Subscription/Process.hs]: Why is this change necessary?

**Comment by etorreborre** [comment]
It looks good to me. Apparently you are also cleaning some of the imports, or it is to make the compilation pass with updated dependencies?

File: tests/Network/Nakadi/Connection/Test.hs [hunk 3]
```diff
@@ -14,6 +14,7 @@ import           Network.Wai.Handler.Warp
 import           System.IO.Unsafe
 import           Test.Tasty
 import           Test.Tasty.HUnit
+import           UnliftIO.Concurrent
```
[the remark this answers is not in the document]
Comment (mtesseract) [reply tests/Network/Nakadi/Connection/Test.hs]: Just checked, looks like it is for bringing 'threadDelay' into scope.

File: tests/Network/Nakadi/EventTypes/Test.hs (hunk 1 above)
Comment (mtesseract) [reply tests/Network/Nakadi/EventTypes/Test.hs]: Yes... This test suite has grown in a more ad-hoc fashion. Would actually be nice to clean this up at some point and provide reasonable building blocks that allow for nicer test writing.

File: tests/Network/Nakadi/Examples/Subscription/Process.hs (hunk 2 above)
Comment (mtesseract) [reply tests/Network/Nakadi/Examples/Subscription/Process.hs]: These are actually two changes. Control.Monad.IO.Unlift was previously required for bringing MonadUnliftIO into scope, which looks like it is now done simply by importing the new ClassyPrelude.

On the other hand, MonadMask is required is required for Nakadi.subscriptionProcess and it seems like it is not exposed anymore by the previously imported modules.

**Comment by mtesseract** [comment]
It was not really my intention to clean up imports, primarily it was simply directed by the compiler errors. :-)
(Not saying that this is not a good idea to be done!)

**Comment by mtesseract** [comment]
Thanks for the review. If you are fine with this, then I will continue with merging.

**Comment by etorreborre** [comment]
Yes, 👍
